streams/src/main/java/org/apache/kafka/streams/state/internals/TimestampedWindowStoreBuilder.java

```diff
@@ -36,7 +36,7 @@ public TimestampedWindowStoreBuilder(final WindowBytesStoreSupplier storeSupplie
                                          final Serde<K> keySerde,
                                          final Serde<V> valueSerde,
                                          final Time time) {
-        super(storeSupplier.name(), keySerde, new ValueAndTimestampSerde<>(valueSerde), time);
+        super(storeSupplier.name(), keySerde, valueSerde == null ? null : new ValueAndTimestampSerde<>(valueSerde), time);
```

Comment:
I found it useful with the FullChangeSerde to make the constructor private and force usage via a null-safe factory method. Then, you never have to worry about this check.

Reply:
Seems to be a single occurrence. Maybe not worth a factory. Also, `ValueAndTimestampSerde` checks for non-null and thus we would surface any issue during testing anyway.